modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy

```diff
@@ -0,0 +1,62 @@
+$version: "2"
+
+namespace alloy.test
+
+use alloy#simpleRestJson
+use smithy.test#httpRequestTests
+use smithy.test#httpResponseTests
+
+apply HttpPayloadWithDefault @httpRequestTests([{
+    id: "SimpleRestJsonSomeHttpPayloadWithDefault"
+    documentation: "Pass simple string value as is if payload provided"
+    protocol: simpleRestJson
+    method: "PUT"
+    uri: "/httpPayloadWithDefault"
+    body: "\"custom value\""
+    headers: {
+        "Content-Type": "application/json"
+    }
+    requireHeaders: ["Content-Length"]
+    params: {
+        body: "custom value"
+    }
+    bodyMediaType: "application/json"
+}, {
+    id: "SimpleRestJsonNoneHttpPayloadWithDefault"
+    documentation: "Use default value when there is no payload"
+    protocol: simpleRestJson
+    method: "PUT"
+    uri: "/httpPayloadWithDefault"
+    headers: {
+        "Content-Type": "application/json"
+    }
```

Comment:
**issue:** hmm I think you'll have to skip the content-type when there's no body: it's not valid JSON to send nothing

Reply:
done